### DISCUSSION TITLE
Linking order

### DISCUSSION THREAD

**Discussion by ivan-pi** [fortran-lang/fpm#462] [General]
I was watching the ` fpm` tutorial from [DJ's Office Hours](https://www.youtube.com/watch?v=pg77uR792YU) and noticed the [manifest reference](https://github.com/fortran-lang/fpm/blob/master/manifest-reference.md#link-external-libraries) says

```
> To dependent on LAPACK also BLAS should be linked. In this case the order of the libraries matters:
[build]
link = ["blas", "lapack"]
```

which if I'm not mistaken is opposite order than if compiling manually (`-llapack -lblas`), since the linker is supposed to satisfy dependencies going from the left to right. (I'm assuming that internally `fpm` adds the flags in reverse order as the one in the TOML file).

Have there been any previous discussions related to selecting the link library order convention? I'm worried that using the "opposite" convention than the linker itself could be confusing for developers familiar with compiling manually.

**Comment by awvwgk** [comment]
The order is not reversed by fpm, but the documentation is in error here. It should read

```toml
[build]
link = ["lapack", "blas"]
```

**Reply by awvwgk** [reply]
Looks like I wasn't paying attention when I wrote the docs to the `build.link` entry, I switched LAPACK and BLAS in all the `link` entries in the manifest reference :sweat_smile:. Good catch.
